pandas/core/groupby/generic.py

```diff
@@ -234,9 +234,16 @@ def aggregate(
         relabeling = func is None
         columns = None
         no_arg_message = "Must provide 'func' or named aggregation **kwargs."
+        tuple_given_message = "func is expected but recieved {} in **kwargs."
         if relabeling:
             columns = list(kwargs)
-            func = [kwargs[col] for col in columns]
+            func = []
+            for col in columns:
+                if isinstance(kwargs[col], (list, NamedAgg, tuple)):
```

Comment:
`NamedAgg` is itself an instance of a tuple, so I don't think it's necessary to include it here

Reply:
Yes, Agreed. Changed them. Pushed new code.